python/lsst/pipe/tasks/computeExposureSummaryStats.py

```diff
@@ -202,94 +149,83 @@ def run(self, exposure, sources, background):
         statObj = afwMath.makeStatistics(exposure.getMaskedImage(), afwMath.STDEVCLIP,
                                          statsCtrl)
         skyNoise, _ = statObj.getResult(afwMath.STDEVCLIP)
+        summary.skyNoise = skyNoise
 
         statObj = afwMath.makeStatistics(exposure.getMaskedImage().getVariance(),
                                          exposure.getMaskedImage().getMask(),
                                          afwMath.MEANCLIP, statsCtrl)
         meanVar, _ = statObj.getResult(afwMath.MEANCLIP)
+        summary.meanVar = meanVar
 
         md = exposure.getMetadata()
         if 'SFM_ASTROM_OFFSET_MEAN' in md:
-            astromOffsetMean = md['SFM_ASTROM_OFFSET_MEAN']
-            astromOffsetStd = md['SFM_ASTROM_OFFSET_STD']
-        else:
-            astromOffsetMean = np.nan
-            astromOffsetStd = np.nan
-
-        psfStats = self._computePsfStats(sources)
-
-        # Note that all numpy values have to be explicitly converted to
-        # python floats for yaml serialization.
-        summary = afwImage.ExposureSummaryStats(
-            psfSigma=float(psfSigma),
-            psfArea=float(psfArea),
-            psfIxx=float(psfIxx),
-            psfIyy=float(psfIyy),
-            psfIxy=float(psfIxy),
-            ra=float(ra),
-            decl=float(decl),
-            zenithDistance=float(zenithDistance),
-            zeroPoint=float(zeroPoint),
-            skyBg=float(skyBg),
-            skyNoise=float(skyNoise),
-            meanVar=float(meanVar),
-            raCorners=raCorners,
-            decCorners=decCorners,
-            astromOffsetMean=astromOffsetMean,
-            astromOffsetStd=astromOffsetStd,
-            nPsfStar=psfStats.nPsfStar,
-            psfStarDeltaE1Median=psfStats.psfStarDeltaE1Median,
-            psfStarDeltaE2Median=psfStats.psfStarDeltaE2Median,
-            psfStarDeltaE1Scatter=psfStats.psfStarDeltaE1Scatter,
-            psfStarDeltaE2Scatter=psfStats.psfStarDeltaE2Scatter,
-            psfStarDeltaSizeMedian=psfStats.psfStarDeltaSizeMedian,
-            psfStarDeltaSizeScatter=psfStats.psfStarDeltaSizeScatter,
-            psfStarScaledDeltaSizeScatter=psfStats.psfStarScaledDeltaSizeScatter
-        )
+            summary.astromOffsetMean = md['SFM_ASTROM_OFFSET_MEAN']
+            summary.astromOffsetStd = md['SFM_ASTROM_OFFSET_STD']
 
         return summary
 
-    def _computePsfStats(self, sources):
-        """Compute psf residual statistics.
-
-        All residuals are computed using median statistics on the difference
-        between the sources and the models.
+    def update_psf(self, summary, psf, bbox, sources=None):
```

Comment:
Will rename to `update_psf_stats` (etc).

I don't have a problem with some amount of case-convention inconsistency within a file, and I think the developer guide provides a lot of leeway on that.  These days I tend to write pretty much any new interface as snake_case, as it's the only way to push us along on that transition even a tiny amount.

Requiring an `exposure` here would indeed be a bit of a problem, as I was hoping not to have to open up the `calexp` at all in the final visit summary stage.  But there's nothing fundamental there; it's just more I/O.  Maybe we limit it to the mask plane (if image plane or variance plane usage in the snippet on DM-36930 adds anything, it means something is wrong with our masking).  Then we could take just the `mask` as an extra parameter to `update_psf_stats`; I could add that here (with it being unused so far) in order to set up a connection for it in the PipelineTask I'll be writing.

Reply:
I've added that `mask` parameter, and have now also factored out the background- and masked-image-based updates into separate methods.  Since I'll be loading the `calexp` mask anyway, it probably makes sense to load those and recompute their stats for consistency.